File: FIC/Linguagens de Programação/JavaScript/sorvete.js
let sabor;

do {
    sabor = parseInt(prompt("Bem-vindo à sorveteria ice!\nPara sair digite 0!\nQual sabor você deseja?\n\n1- Chocolate\n2- Morango\n\nDigite uma opção: "));
    if (sabor != 0 && sabor != 1 && sabor != 2) {
        alert("Opção inválida.\nApenas digite 1 para Chocol ate ou 2 para Morango ou 0 para sair.");
    }
} while (sabor != 0 && sabor != 1 && sabor != 2);

let op;

while (sabor === 1 || sabor === 2) {
    let bolas = parseInt(prompt("Informe a quantidade de bolas: "));
    
    while (isNaN(bolas) || bolas === "" || bolas === 0 || bolas < 0) {
        bolas = parseInt(prompt("Você não informou a quantidade de bolas\nPor favor informe a quantidade de bolas"));
    }
    
    if (sabor === 1) {
        if (bolas > 3) {
            alert("Você tem direito a um desconto de 10%.");
        } else {
            alert("Você tem direito a um desconto de 5%.");
        }
    } else {
        alert("Sorvete sem desconto.");
    }
    
    op = parseInt(prompt("Deseja fazer uma nova compra?\n\n1- Sim\n2- Não\n\nDigite uma opção:"));
    while (op != 1 && op != 2) {
        op = parseInt(prompt("Opção inválida\nDeseja fazer uma nova compra?\n\n1- Sim\n2- Não"));
    }
    w
    if (op === 1) {
        do {
            sabor = parseInt(prompt("Qual sabor você deseja?\n\n1- Chocolate\n2- Morango\n\nDigite uma opção: "));
            if (sabor != 0 && sabor != 1 && sabor != 2) {
                alert("Opção inválida. Digite 1 para Chocolate, 2 para Morango ou 0 para sair.");
            }
        } while (sabor != 0 && sabor != 1 && sabor != 2);
    } else {
        alert("Sistema encerrado");
    }
}

File: FIC/Linguagens de Programação/Java/Sorvete.java
import java.util.Scanner;

class Sorvete {
    public static void main(String[] args) {
        int sabor = 0;
        int bolas = 0;
        int opcao = 0;

        Scanner sorvete = new Scanner(System.in);

        while (true) {
            System.out.print("Bem-vindo à sorveteria ice!\nPara sair, digite 0!\nPor favor escolha um sabor:\n\n1- Chocolate\n2- Morango\n\nDigite uma opção: ");

            try {
                sabor = Integer.parseInt(sorvete.nextLine());

                if (sabor == 0) {
                    System.out.println("Sistema encerrado!");
                    break;
                } else if (sabor >= 1 && sabor <= 2) {
                    System.out.print("Informe a quantidade de bolas: ");
                    
                    while (true) {
                        try {
                            bolas = Integer.parseInt(sorvete.nextLine());

                            if (bolas > 0) {
                                break; // Sair do loop se a quantidade for válida
                            } else {
                                System.out.print("Quantidade inválida. Por favor, digite novamente: ");
                            }
                        } catch (NumberFormatException e) {
                            System.out.print("Quantidade inválida. Por favor, digite novamente: ");
                        }
                    }

                    if (sabor == 1) {
                        if (bolas > 3) {
                            System.out.println("Você tem direito a um desconto de 10%");
                        } else {
                            System.out.println("Você tem direito a um desconto de 5%");
                        }
                    } else {
                        System.out.println("Sorvete sem desconto");
                    }

                    System.out.print("Muito obrigado pela sua compra\nDeseja fazer uma nova compra?\n\n1- SIm\n2- Não\n\nDigite uma opção: ");
                    
                    while (true) {
                        try {
                            opcao = Integer.parseInt(sorvete.nextLine());

                            if (opcao == 1 || opcao == 2) {
                                break; // Sair do loop se a opção for válida
                            } else {
                                System.out.print("Opção inválida. Por favor, digite novamente: ");
                            }
                        } catch (NumberFormatException e) {
                            System.out.print("Opção inválida. Por favor, digite novamente: ");
                        }
                    }

                    if (opcao == 1) {
                        continue; // Continue para a próxima iteração do loop
                    } else {
                        System.out.println("Sistema encerrado!");
                        break;
                    }
                } else {
                    System.out.print("Opção inválida. Por favor, digite novamente:\n");
                }
            } catch (NumberFormatException e) {
                System.out.print("Opção inválida. Por favor, digite novamente:\n");
            }
        }
        sorvete.close();
    }
}
